src/classes/tmDaoProvisioning.cls

```diff
@@ -0,0 +1,174 @@
+                                        ];
+         return trialOrderItems;
+    }
+    public static List<AggregateResult> getTrialOrderItems(Id accountId, Id tenantId) {
+         List<AggregateResult> trialOrderItems = [SELECT Order.ContractId, Pricebookentry.Product2Id, SUM(Quantity)TotalQuantity
+                                        FROM OrderItem
+                                        WHERE Order.IsTrialOrder = true
+                                        AND Order.AccountId = :accountId
+                                        AND ProvisioningTargetId = :tenantId
+                                        AND Order.Status = :tmConstants.ORDER_STATUS_PROVISIONED
+                                        AND Order.Contract.Status = :tmConstants.CONTRACT_STATUS_ACTIVATED                                 
+                                        GROUP BY Order.ContractId, Pricebookentry.Product2Id];
+         return trialOrderItems;
+    }
+
+    public static List<OrderItem> getTrialContracts(Id accountId, Id tenantId) {
+        List<OrderItem> ordItems  = [SELECT Order.ContractId, Order.Contract.EndDate
+                                        FROM OrderItem
+                                        WHERE Order.IsTrialOrder = true
+                                        AND Order.AccountId = :accountId
+                                        AND ProvisioningTargetId = :tenantId
+                                        AND Order.Contract.Status = :tmConstants.CONTRACT_STATUS_ACTIVATED
+                                        ORDER BY Order.Contract.EndDate DESC];
+        return ordItems;
+    }
+
+    public static List<Contract> getCustomerSignedContract(Id accountId, Id tenantId) {
+        List<Contract> customerContracts = new List<Contract>();
+        List<OrderItem> ordItems  = [SELECT Order.ContractId, Order.Contract.StartDate
+                                        FROM OrderItem
+                                        WHERE Order.IsTrialOrder = false
+                                        AND Order.AccountId = :accountId
+                                        AND ProvisioningTargetId = :tenantId
+                                        AND Order.Contract.Status = :tmConstants.CONTRACT_STATUS_ACTIVATED
+                                        AND Order.Contract.StartDate >= :System.Today() 
+                                        ORDER BY Order.Contract.StartDate ASC];
+        if (!ordItems.isEmpty()) {
+            customerContracts.add(ordItems.get(0).Order.Contract);
+        }
+        return customerContracts;
+    }
+
+    public static Map<String, sfbase__CountryRevenueMap__c> getCountryRevMap() {
+        Map<String, sfbase__CountryRevenueMap__c> countryRevMap = new Map<String, sfbase__CountryRevenueMap__c> ();
+        List<sfbase__CountryRevenueMap__c> countryRevList = [SELECT Id, sfbase__CountryISOCode__c, sfbase__CountryName__c, sfbase__PrimaryCurrency__c, sfbase__RevenueOwner__c
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.